mlrun/secrets.py

```diff
@@ -264,6 +265,11 @@ def sync_secret_tokens() -> None:
     # TODO: Runtime Context Check - Avoid sending a backend request when running inside a runtime, where secrets
     #  are already injected via Kubernetes and syncing is unnecessary
 
+    # Skip syncing secrets when running inside a job,
+    # since the offline token is already provided via the environment variable
+    if os.getenv("MLRUN_AUTH_OFFLINE_TOKEN"):
```

Comment:
I think this heuristic of assuming running as mlrun job is wrong, need better logic